expression/aggregation/base_func.go

```diff
@@ -411,6 +411,9 @@ func (a *baseFuncDesc) WrapCastForAggArgs(ctx sessionctx.Context) {
 		if i == 1 && (a.Name == ast.WindowFuncLead || a.Name == ast.WindowFuncLag || a.Name == ast.WindowFuncNthValue) {
 			continue
 		}
+		if a.Args[i].GetType().Tp == mysql.TypeNull {
```

Comment:
Why add this check?

Reply:
If it casts `null` as int. The return type would be inferred as `decimal` when building aggregation. But the schema is int (correct type). Which makes execution panic when fetching the results.

Reply:
why casts null as int will cause return type to be infereed as decimal? would you please point it out (I am not clear about this, and I didn't found it myself, please let me know)

Reply:
You can see `func AggFieldType(tps []*FieldType) *FieldType`